internal/impl/sftp/input.go

```diff
@@ -242,11 +241,22 @@ func (s *sftpReader) seekNextPath(ctx context.Context) (file *sftp.File, nextPat
 		s.pathProvider = s.getFilePathProvider(ctx)
 	}
 
+	return s.client, s.pathProvider, false, nil
```

Comment:
As a newcomer to this code this feels unsafe.

If the only important part of this code that all these variables are accessed/set together atomically, I do wonder if an atomic is better suited. You can use `Swap` to set the new value and destroy in `Close`, `Store` in Connect and `Load` in ReadBatch. I don't quite understand the higher level contract here of **why** it's only required that they are accessed concurrently and we don't have to worry about `Close` clobbering something ongoing in `Connect` or `ReadBatch`.

Reply:
And when I talk about using atomics I mean using `typed.AtomicValue` in our typed package in `internal/typed` and wrapping all this state into a struct.